src/Nethermind/Nethermind.Consensus.AuRa/Validators/ContractBasedValidator.cs

```diff
@@ -94,12 +95,12 @@ public override void OnBlockProcessingStart(Block block, ProcessingOptions optio
             bool isInitBlock = InitBlockNumber == block.Number;
             bool isProducingBlock = options.ContainsFlag(ProcessingOptions.ProducingBlock);
             bool isMainChainProcessing = !ForSealing && !isProducingBlock;
-            bool isConsecutiveBlock = block.Number - 1 == _lastProcessedBlockNumber && _lastProcessedBlockNumber != 0;
+            bool isInProcessedRange = _lastProcessedBlockNumber is not null && block.Number - 1 <= _lastProcessedBlockNumber;
```

Comment:
why `<=` ? If it is not next block shouldn't we always load validators from contract? If we go back 100 blocks for example?

Reply:
`_lastProcessedBlockNumber` is an in-memory field, so if it is not null it means that we have processed, during this run, up to the block it is pointing to, thus we would have stored validators to the validator store database. Loading validators from the contract here is potentially safer, but it breaks unit tests (which we could fix ofc), but worth noting that this is the current behavior on master, not something introduced here, I just renamed variable to `inProcessedRange` for clarity.